gensim/models/fast_knn.py

```diff
@@ -0,0 +1,224 @@
+        """
+        self.vectorizer = CountVectorizer(vocabulary = common_vocabulary, dtype = np.double)
+        self.docs = normalize(check_array([doc.toarray().ravel() for doc in self.vectorizer.transform(docs)]), norm='l1') # BOW : Normalised Bag Of Words representation of doc
+        self.id2doc = dict([(idx, doc) for idx, doc in enumerate(self.docs)]) # creates doc_index mapping of docs to docs
+        
+    def _create_embedding(self, docs):
+        """
+        This function creates a word2vec model over the given data and creates an embedding matrix (self.word_embedding)
+        of dimension (num of docs X dimension of word2vec vector)
+
+        It returns the tokens for BOW
+
+        Parameters
+        ----------
+
+        docs : Input docs conveted to their BOW representation.(no. of docs X vocabulary size)
+        """
+        vectorizer = CountVectorizer(stop_words = "english").fit(docs)
+        tokenizer = vectorizer.build_tokenizer() # sklearn tokenizer to break the sentence into tokens
+        
+        word_embedding_model = Word2Vec([tokenizer(doc) for doc in docs], min_count = 1) # creating word2vec model from the given data
+        model_vocab = word_embedding_model.vocab
+        common_vocab = [word for word in vectorizer.get_feature_names() if word in model_vocab]
+        self.word_embedding = check_array([word_embedding_model[word] for word in common_vocab])  # representation of tokens in model space : no of docs X word2vec vector size
+        self.word_embedding_distance = euclidean_distances(self.word_embedding)
+        # np.fill_diagonal(self.word_embedding_distance, float('inf'))
+    #         print "fdsd", self.word_embedding.shape
+        return common_vocab
+
+    def _pairwise_wcd_dist_row(self, test_doc):
+        """
+        This function returns the sorted word centroid distances of all given docs with respect to query doc.
+        
+        Parameters
+        ----------
+
+        test_doc : a normalised BOW representation of the query doc.
+        """        
+        wcd_distances = Parallel(n_jobs = self.n_jobs)(
+                                delayed(self.get_wcd)(test_doc, self.docs[doc_id], doc_id)
+                                for doc_id in range(len(self.docs)))
+
+        return sorted(dict(wcd_distances).items(), key = itemgetter(1))
+
+    def get_wcd(self, test_doc, doc, doc_id):
+    #         print self.word_embedding.shape
+    #         print doc1.shape
+    #         print doc2.shape
+        """
+        This function calculates the Word Centroid Distance between docs
+        
+        Parameters
+        ----------
+
+        test_doc : Normalised BOW representaion of test doc .
+        doc : Normalised BOW representaion of stored doc .
+        doc_id : id of the stored doc
+        """
+        return (doc_id, distance.euclidean(np.dot(np.transpose(self.word_embedding), np.transpose(test_doc)), np.dot(np.transpose(self.word_embedding), doc)))
```

Comment:
this is because doc is an array containing the reserved documents and the doc_id is the index of a document in the doc array. I should have named docs as docs_arr. I will correct it in next commit.
So, Continuing when I access doc[doc_id] i get an array of shape (1,x) while test_doc being an array has shape (x,). Now dimensions of word embedding is (x,1).
So, to make the shapes compatible, transpose of test_doc is  taken, to make it (1,x)